### PR TITLE
added trending petition sort

### PR DESCRIPTION
**Pull request by jukart**

### CODE REVIEW

File: iris/service/content/petition/service.py [hunk 1]
```diff
@@ -186,6 +191,89 @@ def _prepare_document(self, doc, data, is_create):
...
+        """Extends the search endpoint with the special `trending` sort.
+
+        It is possible to provide the sort name `trending` which will then
+        sort by trending petitions.
+        If this sort is used all parameters except the `limit` are ignored.
+
+        This is implemented by looking up for the petitions with the most
+        supports in the last `look back days` (see DEFAULT_TRENDING).
+        If the query provides less than `limit` petitions the time range is
+        extended by `increment`. This repeated until `max look back days` is
```
Comment (lukasender) [inline iris/service/content/petition/service.py]: This `is` repeated

File: iris/service/rest/queries.py [hunk 2]
```diff
@@ -79,6 +79,23 @@ def genericTermsQuery(value):
     }
 
 
+def rangeFilter(field, operators=('gte', 'lt')):
+    def handleFilter(value):
+        operations = {}
+        for i, op in enumerate(operators):
+            if i > len(value):
+                break
+            v = value[i]
```
Comment (lukasender) [inline iris/service/rest/queries.py:88]: in the next line, you're making sure that "v" is "truthy". can `value[i]` every raise a `KeyError`?
If so and if you do not want to fail hard here, you might want to consider changing this to `value.get(i, None)`

File: iris/service/content/petition/service.py [hunk 3]
```diff
@@ -186,6 +191,89 @@ def _prepare_document(self, doc, data, is_create):
...
+               sort=None,
+               testing_only=False,
+               debug=None,
+               trending=None,
+               **params):
+        """Extends the search endpoint with the special `trending` sort.
+
+        It is possible to provide the sort name `trending` which will then
+        sort by trending petitions.
+        If this sort is used all parameters except the `limit` are ignored.
```
Comment (lukasender) [inline iris/service/content/petition/service.py:211]: Maybe be more explicit here?

```
If `sort='trending'` is used...
```

File: iris/service/content/petition/service.py [hunk 4]
```diff
@@ -186,6 +191,89 @@ def _prepare_document(self, doc, data, is_create):
...
+
+        This is implemented by looking up for the petitions with the most
```
Comment (lukasender) [inline iris/service/content/petition/service.py]: Sounds a little bit off

How about?

```
Trending petitions is implemented by searching for petitions with the most supports within the last `look ba...
```

**Review by lukasender — COMMENTED** [review]
consider the above, otherwise ack

File: iris/service/rest/queries.py (hunk 2 above)
Comment (jukart) [reply iris/service/rest/queries.py:88]: It is checked in the lines above

**Pull request merged**